internal/lifecycle/lifecycle.go

```diff
@@ -38,17 +39,19 @@ type Hook struct {
 
 // Lifecycle coordinates application lifecycle hooks.
 type Lifecycle struct {
-	logger     *fxlog.Logger
+	logger     fxlog.Logger
 	hooks      []Hook
 	numStarted int
 }
 
 // New constructs a new Lifecycle.
-func New(logger *fxlog.Logger) *Lifecycle {
+func New(logger fxlog.Logger) *Lifecycle {
 	if logger == nil {
-		logger = fxlog.New()
+		logger = fxlog.DefaultLogger(os.Stderr)
 	}
```

Comment:
i know that the old code did this, but i would drop the default logge so that we don't accidentally write to stderr.